tests/30rooms/13anonymousaccess.pl

```diff
@@ -0,0 +1,267 @@
+                     dir => "b",
+                  },
+               )->followed_by( \&main::expect_http_400 )->then( sub {
+                  do_request_json_for( $anonymous_user,
+                     method => "GET",
+                     uri => "/api/v1/events",
+                     params => {
+                        limit => "2",
+                        dir => "b",
+                        room_id => $room_id,
+                     },
+                  )
+               })->then( sub {
+                  my ( $body ) = @_;
+
+                  require_json_keys( $body, qw( chunk ) );
+                  $body->{chunk} >= 1 or die "Want at least one chunk";
+                  my $chunk = $body->{chunk}[0];
+                  require_json_keys( $chunk, qw( content ) );
+                  my $content = $chunk->{content};
+                  require_json_keys( $content, qw( body ) );
+                  $content->{body} eq "mice" or die "Want content body to be mice";
+
+                  Future->done( 1 );
+               }),
+            );
+         });
+      });
+   };
+
+test "Anonymous user doesn't get events before room made world_readable",
+   requires => [ qw( first_api_client ), local_user_preparer() ],
+
+   do => sub {
+      my ( $api_client, $user ) = @_;
+
+      my $anonymous_user;
+      my $room_id;
+
+      register_anonymous_user( $api_client )->then( sub {
+         ( $anonymous_user ) = @_;
+
+         matrix_create_and_join_room( [ $user ] )
+         ->then( sub {
+            ( $room_id ) = @_;
+
+            Future->needs_all(
+               delay( 0.05 )->then( sub {
+                  matrix_send_room_text_message( $user, $room_id, body => "private" )->then(sub {
+                     do_request_json_for( $user,
+                        method => "PUT",
+                        uri    => "/api/v1/rooms/$room_id/state/m.room.history_visibility/",
+
+                        content => {
+                           history_visibility => "world_readable",
+                        },
+                     );
+                  })->then( sub {
+                     matrix_send_room_text_message( $user, $room_id, body => "public" );
+                  });
+               }),
+
+               check_events( $anonymous_user, $room_id )
+               ->then(sub {
+                  Future->done( 1 );
+               }, sub {
+                  check_events( $anonymous_user, $room_id );
+               }),
+            );
+         });
+      });
+   };
+
+sub check_events
+{
+   my ( $user, $room_id ) = @_;
+
+   do_request_json_for( $user,
+      method => "GET",
+      uri => "/api/v1/events",
+      params => {
+         limit => "3",
+         dir => "b",
+         room_id => $room_id,
+      },
+   )->then( sub {
+      my ( $body ) = @_;
+
+      log_if_fail "Body", $body;
+
+      require_json_keys( $body, qw( chunk ) );
+      @{$body->{chunk}} >= 1 or die "Want at least one chunk";
```

Comment:
The chunk is an array of events, this should probably read ``"Want at least one event"``.